tools/FlatFlowMigrationCli/Operations/MigrateOperation.cs

```diff
@@ -0,0 +1,267 @@
+
+        await MigrateIncomingSubscriptions(sourceMappings, incomingSubscriptions);
+        await MigrateOutgoingSubscriptions(sourceMappings, repoUri, outgoingSubscriptions);
+
+        var arcadeSubscription = incomingSubscriptions
+            .FirstOrDefault(s => s.SourceRepository == Constants.ArcadeRepoUri);
+
+        // If we depend on older Arcade, we want to exclude it during backflows
+        HashSet<string> excludedAssets = arcadeSubscription != null && arcadeSubscription.Channel.Name != Constants.LatestArcadeChannel
+            ? await GetArcadePackagesToExclude(branch, repoUri, arcadeSubscription)
+            : [];
+
+        await _subscriptionMigrator.CreateBackflowSubscriptionAsync(dependency.Mapping.Name, repoUri, branch, excludedAssets);
+
+        _logger.LogInformation("Repository {mapping} successfully migrated", dependency.Mapping.Name);
+    }
+
+    private async Task VerifyNoPatchesLeft(VmrDependency dependency)
+    {
+        try
+        {
+            var sdkMainSha = await _gitHubClient.GetLastCommitShaAsync(Constants.SdkRepoUri, "main")
+                 ?? throw new InvalidOperationException($"Failed to get the latest SHA of the main branch of {Constants.SdkRepoUri}");
+            var files = await _gitHubClient.GetFilesAtCommitAsync(Constants.SdkRepoUri, sdkMainSha, Constants.SdkPatchLocation);
+            var repoPatchPrefix = $"{Constants.SdkPatchLocation}/{dependency.Mapping.Name}/";
+
+            if (files.Any(f => f.FilePath.StartsWith(repoPatchPrefix)))
+            {
+                throw new InvalidOperationException($"Repository {dependency.Mapping.Name} still has source build patches in dotnet/sdk!");
+            }
+        }
+        catch (Exception e) when (e.Message.Contains("could not be found"))
+        {
+            // No patches left in dotnet/sdk
+        }
+    }
+
+    /// <summary>
+    /// Migrates subscriptions originating in the currently migrated repository.
+    /// </summary>
+    private async Task MigrateOutgoingSubscriptions(
+        IReadOnlyCollection<SourceMapping> sourceMappings,
+        string repoUri,
+        List<Subscription> outgoingSubscriptions)
+    {
+        var vmrChannel = (await _pcsClient.Channels.ListChannelsAsync())
+            .First(c => c.Name == Constants.VmrChannelName);
+
+        foreach (var outgoing in outgoingSubscriptions)
+        {
+            _logger.LogInformation("Processing outgoing subscription {subscriptionId} {sourceRepository} -> {targetRepository}...",
+                outgoing.Id,
+                outgoing.SourceRepository,
+                outgoing.TargetRepository);
+
+            var targetIsInVmr = sourceMappings.Any(m => m.DefaultRemote == outgoing.TargetRepository);
+            if (!targetIsInVmr && IsVmrRepoWithOwnOfficialBuild(repoUri))
+            {
+                _logger.LogInformation("Skipping subscription {subscriptionId} as parent repo has own official build and dependents will stay subscribed there",
```

Comment:
Are we officially calling the product repo a `parent repo`?

Reply:
This is with regards to the dependency tree

Reply:
Do you think it needs to be rewritten?

Reply:
nah, think it's fine